src/main/java/com/yugabyte/sample/apps/MultiTableMultiIndexInserts.java

```diff
@@ -0,0 +1,449 @@
+// Copyright (c) YugaByte, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.  You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software distributed under the License
+// is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+// or implied.  See the License for the specific language governing permissions and limitations
+// under the License.
+//
+package com.yugabyte.sample.apps;
+
+import java.sql.Array;
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.Statement;
+import java.util.*;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.concurrent.atomic.AtomicLong;
+
+import org.apache.log4j.Logger;
+
+/**
+ * This workload writes and reads some random string keys from a postgresql table.
+ */
+public class MultiTableMultiIndexInserts extends AppBase {
+  private static final Logger LOG = Logger.getLogger(MultiTableMultiIndexInserts.class);
+
+  // Static initialization of this workload's config.
+  static {
+    appConfig.readIOPSPercentage = -1;
+    appConfig.numReaderThreads = 0;
+    appConfig.numWriterThreads = 2;
+    appConfig.numKeysToRead = -1;
+    appConfig.numKeysToWrite = -1;
+  }
+
+  private static final AtomicInteger atomicInt = new AtomicInteger(0);
+  private static final AtomicLong atomicLong = new AtomicLong(0);
+
+  class PerThreadState {
+    private  Map<String, PreparedStatement> preparedStatements = new HashMap<>();
+    private Connection connection;
+    private Random random = new Random();
+
+    // The following variables are cached from the global incrementing variable so that we don't need to
+    // use an expensive atomic operation to get a single value. We cache 'kAtomicDelta' ints at a time.
+    private int nextInt;
+    private int nextIntMax;
```

Comment:
I think the "counter caching" code would be a bit more straightforward if `nextIntMax` and `nextLongMax` bounds were exclusive, i.e. one higher than the current "max" values. They could be renamed to something `nextUncachedInt` and `nextUncachedLong`.